sbi/inference/potentials/score_based_potential.py

```diff
@@ -0,0 +1,196 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
+
+from typing import Optional, Tuple
+
+import torch
+from torch import Tensor
+from torch.distributions import Distribution
+
+from sbi.inference.potentials.base_potential import BasePotentialGradient
+from sbi.neural_nets.estimators.score_estimator import ConditionalScoreEstimator
+from sbi.sbi_types import TorchTransform
+from sbi.utils import mcmc_transform
+
+
+def score_estimator_based_potential_gradient(
+    score_estimator: ConditionalScoreEstimator,
+    prior: Optional[Distribution],
+    x_o: Optional[Tensor],
+    enable_transform: bool = False,
+) -> Tuple["ScoreFunction", TorchTransform]:
+    r"""Returns the potential function gradient for score estimators.
+
+    Args:
+        score_estimator: The neural network modelling the score.
+        prior: The prior distribution.
+        x_o: The observed data at which to evaluate the score.
+        enable_transform: Whether to enable transforms. Not supported yet.
+
+    """
+    device = str(next(score_estimator.parameters()).device)
+
+    potential_fn = ScoreFunction(score_estimator, prior, x_o, device=device)
+
+    # TODO Add issue
+    assert (
+        enable_transform is False
+    ), "Transforms are not yet supported for score estimators."
+
+    if prior is not None:
+        theta_transform = mcmc_transform(
+            prior, device=device, enable_transform=enable_transform
+        )
+    else:
+        theta_transform = torch.distributions.transforms.identity_transform
+
+    return potential_fn, theta_transform
+
+
+class ScoreFunction(BasePotentialGradient):
```

Comment:
Can be a bit confusing mixing between `PotentialGradient` and `Score` - these are not the same thing. Maybe we can rename `BasePotentialGradient`?